Bootloader/init.cpp

```diff
@@ -21,15 +21,16 @@ extern "C" [[noreturn]] void __stack_chk_fail();
 extern "C" u8 start_of_bootloader_image[];
 extern "C" u8 end_of_bootloader_image[];
 
-extern "C" void* gdt64ptr;
-extern "C" __attribute__((section(".boot_text"))) void* code64_sel;
+extern "C" u8 gdt64ptr[];
+extern "C" u16 code64_sel;
 extern "C" u64 boot_pml4t[512];
 extern "C" u64 boot_pdpt[512];
 extern "C" u64 boot_pd0[512];
+extern "C" u64 boot_pd0_pts[512 * (MAX_KERNEL_SIZE >> 21)];
 extern "C" u64 boot_pd_kernel[512];
-extern "C" u64 boot_pd_kernel_pts[512 * (KERNEL_PD_OFFSET >> 21)];
+extern "C" u64 boot_pd_kernel_pts[512 * (MAX_KERNEL_SIZE >> 21)];
```

Comment:
Maybe &1FF here?
Same in the boot.S file

Reply:
Fixed. :)